platform/core/src/services/ColorbarService/ColorbarService.ts

```diff
@@ -0,0 +1,146 @@
+import { PubSubService } from '../_shared/pubSubServiceInterface';
+
+const EVENTS = {
+  COLORBAR_STATE_CHANGED: 'event::ColorbarService:stateChanged',
+};
+
+export default class ColorbarService extends PubSubService {
+  colorbars = {};
+  public static REGISTRATION = {
+    name: 'colorbarService',
+    create: () => {
+      return new ColorbarService();
+    },
+  };
+
+  constructor() {
+    super(EVENTS);
+  }
+
+  addColorbar(viewportId, element, options = {}, ViewportColorbar) {
```

Comment:
if you have viewportId, then you have the element (can grab it from the renderingEngine), so please remove element, 

Also the options being devault = {} and after that another argument is unusual
I expect `addColorbar(viewportId, colorbarObjectOrWhatever, options)`

Reply:
Done